security/fma-holocene.md

```diff
@@ -0,0 +1,196 @@
+  - [Derivation Deadlock (either specification or op-node bug)](#derivation-deadlock-either-specification-or-op-node-bug)
+    - [Description](#description-1)
+    - [Risk Assessment](#risk-assessment-1)
+    - [Mitigations](#mitigations-1)
+    - [Detection](#detection-1)
+    - [Recovery Path(s)](#recovery-paths-1)
+- [Configurable EIP-1559 Parameters via SystemConfig](#configurable-eip-1559-parameters-via-systemconfig)
+  - [Risk Assessment](#risk-assessment-2)
+  - [Mitigations](#mitigations-2)
+- [Update to the MIPS contract](#update-to-the-mips-contract)
+  - [Undetected go1.22 changes](#undetected-go122-changes)
+    - [Risk Assessment](#risk-assessment-3)
+    - [Mitigations](#mitigations-3)
+    - [Detection](#detection-2)
+  - [Divergent FPVM implementations](#divergent-fpvm-implementations)
+    - [Risk Assessment](#risk-assessment-4)
+    - [Mitigations](#mitigations-4)
+    - [Detection](#detection-3)
+- [Generic Items](#generic-items)
+- [Audit Requirements](#audit-requirements)
+- [Action Items](#action-items)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+|                     |             |
+| ------------------- | ----------- |
+| Author              | George Knee, Sebastian Stammler, Roberto Boyardo, Mofi Taiwo |
+| Created at          | 2024-09-24  |
+| Needs Approval From | maurelian   |
+| Other Reviewers     |             |
+| Status              | Draft       |
+
+## Introduction
+
+This document is intended to be shared in a public space for reviews and visibility purposes. It covers the Holocene hardfork, which involves the following changes:
+
+- (Consensus Layer) Holocene Derivation
+- (Execution Layer & Smart Contracts) Configurable EIP-1559 Parameters via SystemConfig
+- (Smart Contracts) Update to the MIPS contract
+
+Each change has it's own section below with a list of Failure Modes.
+
+Below are references for this project:
+
+- [PID: Holocene hardfork upgrade](https://www.notion.so/PID-Holocene-hardfork-upgrade-00ee1ffc414a407088fdb49841771527?pvs=21)
+- [Github tracker](https://github.com/orgs/ethereum-optimism/projects/84/views/6)
+- [Specs](https://specs.optimism.io/protocol/holocene/derivation.html?highlight=holocene#holocene-derivation)
+
+## Holocene Derivation
+
+### The batcher violates the new (stricter) ordering rules
+
+#### Description
+
+If the batcher:
+
+- sends transactions out of order, or
+- orders frames within transactions out of order, or
+- orders batches within a channel out of order
+
+This will lead to the consensus client hitting error conditions when it loads the frames from L1. According to the spec, frames will be dropped in order to maintain a contiguous frame queue, and out-of-order batches will be dropped at a later stage in the derivation pipeline. The safe chain would halt until the next batch can be resubmitted in order.
+
+#### Risk Assessment
+
+low severity / low likelihood
+
+#### Mitigations
+
+The batcher implementation could:
```

Comment:
I still think the right abstraction for the batcher is to read an API that exposes the internal state of the derivation pipeline and it's able to simulate batches against a local derivation pipeline within the batcher itself before submitting the batch. Obviously this is a lot of work and out of scope but curious if people think this is a good idea

Reply:
@tynes let's continue this convo, but I am going to resolve this in the interest of wrapping up the FMA>